uk.co.compucorp.civicrm.tasksassignments/js/test/services/settings.spec.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'mocks/data/app-settings',
+  'tasks-assignments/app'
+], function (_, settingsMock) {
+  'use strict';
+
+  var mockBackendCalls = function ($httpBackend) {
```

Comment:
1) Why is that all needed? Because of `AppSettings.get()`?
2) Is it possible to mock requests responses with a proper data instead of `{}`?

Reply:
> 1. Why is that all needed? Because of AppSettings.get()?

Its not because of  `AppSettings.get()`. Its because when the angular app bootstraps it attempts to make api calls to get and store some data like status, options, etc for future use. So to mock the api calls we are using `mockBackendCalls` function to mock possible api calls in advance. 

> 2. Is it possible to mock requests responses with a proper data instead of {}?

Of course its possible! but we wont be needing those responses to test  this service. So, i think its ok to mock the response as `{}`.

Reply:
In that case, can we just simply squash these lines somehow
```js
$httpBackend.whenGET(/action=*/).respond({});
```
and then override whichever we need? will it work?

Reply:
good idea! fixed!